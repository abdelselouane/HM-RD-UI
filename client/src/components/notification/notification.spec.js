/* describe, it */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Notification from './index';

describe('<Notification />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      title: 'mockTitle',
      className: 'mock-class',
      onClick: jest.fn(),
      isOpen: true
    };

    wrapper = shallow(<Notification {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render message content if isOpen is true', () => {
    wrapper.setProps({ isOpen: true });
    expect(wrapper.find('.banner.open .message')).toHaveLength(1);
  });

  it('should not render message content if isOpen is false', () => {
    wrapper.setProps({ isOpen: false });
    expect(wrapper.find('.banner.open .message')).toHaveLength(0);
  });

  it('should call button function on click', () => {
    const spy = jest.spyOn(props, 'onClick');
    wrapper.setProps({ isOpen: true });
    wrapper.find('.close').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
