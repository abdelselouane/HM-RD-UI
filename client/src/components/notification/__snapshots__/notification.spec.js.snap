// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should render properly 1`] = `
<div
  className="banner mock-class open"
>
  <span
    className="message"
  >
    mockTitle
  </span>
  <span
    className="close"
    onClick={[Function]}
  />
</div>
`;
